Update COMException to contain HRESULT and introduce COMInvokeException

matthiasblaesing opened the pull request:
Please see the details of the commits.

dblock commented on contrib/msoffice/src/com/sun/jna/platform/win32/COM/office/MSOfficeDemo.java:
```diff
@@ -103,9 +104,10 @@ public void testMSWord() throws IOException {
-            if (e.getExcepInfo() != null) {
-                System.out.println("bstrSource: " + e.getExcepInfo().bstrSource);
-                System.out.println("bstrDescription: " + e.getExcepInfo().bstrDescription);
+            if (e instanceof COMInvokeException) {
```
This should just be a dedicated catch of `COMInvokeException` and there should be a catch for `COMException` after it that just gets ignored.

dblock commented on contrib/platform/src/com/sun/jna/platform/win32/COM/COMException.java:
```diff
@@ -35,93 +33,74 @@
-
-    public void setuArgErr(int uArgErr) {
-        this.uArgErr = uArgErr;
+    public boolean matchesHresult(int errorToCheck) {
```
This seems to imply that the argument is the Hresult. Should be `matches` or `matchesErrorCode`.

matthiasblaesing commented:
@dblock Thanks for looking into this. I adjusted the commits, for the msoffice sample I completely agree, for the "COMException#matchesErrorCode" I have no strong feelings either way. I think the javadoc was ok, but I adjusted the method name and am ok with it.
